src/components/external-dns/index.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  Fragment,
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+
+import * as PropTypes from 'prop-types';
+import { Checkbox, Icon, Table, Typography } from '@equinor/eds-core-react';
+import { ExternalDns, Tls } from '../../store/radix-api';
+import { ExternalDNSStatusBadge } from '../status-badges';
+import { chevron_down, chevron_up } from '@equinor/eds-icons';
+import clsx from 'clsx';
+import { TLSCertificateList } from '../tls-certificate-list';
+import { Alert, AlertProps } from '../alert';
+import { differenceInDays } from 'date-fns';
+import { pluraliser } from '../../utils/string';
+import { dataSorter, sortCompareString } from '../../utils/sort-utils';
+
+const dayPluraliser = pluraliser('day', 'days');
+
+function useGetSortedExternalDNS(
+  externalDnsList: Array<ExternalDns>
+): Array<ExternalDns> {
+  const [sortedData, setSortedData] = useState(externalDnsList);
+
+  useEffect(() => {
+    setSortedData(
+      dataSorter(externalDnsList, [(x, y) => sortCompareString(x.fqdn, y.fqdn)])
+    );
+  }, [externalDnsList]);
+
+  return sortedData;
+}
+
+const AlertTemplates: Record<Tls['status'], AlertProps> = {
+  Pending: { type: 'info' },
+  Consistent: { type: 'info' },
+  Invalid: { type: 'danger' },
+};
+
+const StatusMessages: FunctionComponent<{
+  status: Tls['status'];
+  messages: Array<string>;
+}> = ({ status, messages }) => (
+  <Alert {...AlertTemplates[status]}>
```

Comment:
Consider a simpler approach: 
```suggestion
type Props = {
  status: Tls['status'],
  messages: Array<string>,
}
function StatusMessages({ status, messages }: Props){
return (
```
(This is also easier and faster for Typescript to parse)

Reply:
fixed